File: desert/basin-stampede.lua
loop
    local enemy = self:findNearest(self:findEnemies())
    local xPos = self.pos.x + 5
    local yPos = 17
    if enemy then
        if enemy.pos.y > self.pos.y then
            yPos = yPos - 3
        elseif enemy.pos.y < self.pos.y then
            yPos = yPos + 3
        end
    end
    self:move({x=xPos, y=yPos})
end

File: desert/basin-stampede.clj
(while true
  (let [e (.findNearest this (.findEnemies this))
        xPos (+ (.x (.pos this)) 5)]
    (def yPos 17)
    (if e
      (if (> (.y (.pos e)) (.y (.pos this)))
        (def yPos 14)
        (if (< (.y (.pos e)) (.y (.pos this)))
          (def yPos 20))))
    (.move this (clj->js { :x xPos, :y yPos }))))
